src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -357,6 +357,33 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
     throw failures.failure(info, shouldContain(actual, entries, notFound));
   }
 
+  /**
+   * Asserts that the given {@code Map} contains the entries of the other {@code Map}, in any order.
+   *
+   * @param <K> key type
+   * @param <V> value type
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Map}.
+   * @param other the {@code Map} of entries that are expected to be in the given {@code Map}.
+   * @throws NullPointerException if the other map is {@code null}.
+   * @throws AssertionError if the given {@code Map} is {@code null}.
+   * @throws AssertionError if the given {@code Map} does not contain the entries of the other {@code Map}.
+   */
+  public <K, V> void assertContainsAllEntriesOf(AssertionInfo info, Map<K, V> actual,
+                                                Map<? extends K, ? extends V> other) {
+    failIfNull(other);
+    assertNotNull(info, actual);
+    // if map of values is empty, then assertion passes.
+    if (other.isEmpty()) return;
+    Set<? extends Map.Entry<? extends K, ? extends V>> entries = other.entrySet();
+    Set<Map.Entry<? extends K, ? extends V>> notFound = new LinkedHashSet<>();
+    for (Map.Entry<? extends K, ? extends V> entry : entries) {
+      if (!containsEntry(actual, entry)) notFound.add(entry);
+    }
+    if (notFound.isEmpty()) return;
+    throw failures.failure(info, shouldContain(actual, entries, notFound));
```

Comment:
To avoid code duplication, extract this section as a private method and use it in `assertContains`